portal/src/graphql/portal/AuthenticationCountryCallingCodeList.tsx

```diff
@@ -0,0 +1,490 @@
+}
+
+interface CountryCallingCodeListItem extends IObjectWithKey {
+  key: string;
+  selected: boolean;
+  countryName: string;
+  callingCode: string;
+}
+
+type CountryCallingCodeListData = Record<
+  string,
+  { key: string; countryName: string; callingCode: string }
+>;
+
+interface CountryCallingCodeListItemCheckboxProps extends ICheckboxProps {
+  index?: number;
+  onCheckboxClicked: (index: number, checked: boolean) => void;
+}
+
+interface CountryCallingCodeListSelectAllProps extends ICheckboxProps {
+  isPartiallySelected: boolean;
+  isAllSelected: boolean;
+  selectAll: () => void;
+  unselectAll: () => void;
+}
+
+const HEADER_STYLE = {
+  check: {
+    width: "35px !important",
+    opacity: "1 !important",
+    paddingLeft: "15px !important",
+  },
+};
+
+const SEARCH_BOX_STYLE = {
+  root: { width: "300px" },
+};
+
+// instantiate fuzzy searcher
+const fuseSearcher = new Fuse<Omit<CountryCallingCodeListItem, "selected">>(
+  [],
+  {
+    shouldSort: false,
+    keys: ["countryName", "callingCode"],
+    // lower means more strict (smaller normalized Levenshtein distance)
+    threshold: 0.4,
+    // setting a large distance means location constraint will be
+    // satisfied no matter where the match is located
+    distance: 10000,
+  }
+);
+
+function makeCountryCodeListColumns(
+  renderToString: (messageId: string) => string
+): IColumn[] {
+  return [
+    {
+      key: "selected",
+      fieldName: "selected",
+      name: renderToString("AuthenticationWidget.phone.list-header.active"),
+      minWidth: 90,
+      maxWidth: 90,
+      className: styles.callingCodeListColumn,
+    },
+    {
+      key: "countryName",
+      fieldName: "countryName",
+      name: renderToString(
+        "AuthenticationWidget.phone.list-header.country-or-area"
+      ),
+      minWidth: 240,
+      maxWidth: 240,
+      isMultiline: true,
+      className: cn(styles.countryNameCell, styles.callingCodeListColumn),
+    },
+    {
+      key: "callingCode",
+      fieldName: "callingCode",
+      name: renderToString("AuthenticationWidget.phone.list-header.code"),
+      minWidth: 65,
+      maxWidth: 65,
+      className: styles.callingCodeListColumn,
+    },
+    {
+      key: "order",
+      name: renderToString("AuthenticationWidget.phone.list-header.order"),
+      fieldName: "order",
+      minWidth: 175,
+      maxWidth: 175,
+      className: styles.callingCodeListColumn,
+    },
+  ];
+}
+
+// asusume country calling code data has no duplicated code
+function constructCallingCodeListData(
+  allCountryCallingCodes: string[],
+  getTelecomCountryName: (code: string) => string
+): CountryCallingCodeListData {
+  const callingCodeListData: CountryCallingCodeListData = {};
+  for (const callingCode of allCountryCallingCodes) {
+    const countryCodes =
+      countryCallingCodeMap[callingCode as keyof typeof countryCallingCodeMap];
+    const countryName = countryCodes
+      .map((code) => getTelecomCountryName(code))
```

Comment:
Let's just use the first country code, otherwise for some entries it gets too long.

Reply:
We can use the first country code for now. But I think we need to come up with a better solution. For example, country code 1 is used by both US and Canada and other areas, only showing one could be confusing.

Also, country code 42 is for Czechoslovakia, I think it should be removed from backend + portal, cannot get country name as the country itself no longer exist.

Reply:
Sure, please help delete it from `country_calling_code.go` as well.

Reply:
pushed 1 more commit to remove it